packages/react/src/components/sideNavigation/mainLevel/MainLevel.tsx

```diff
@@ -0,0 +1,198 @@
+  active,
+  children,
+  className,
+  href,
+  icon,
+  id,
+  index,
+  label,
+  withDivider,
+  external,
+  openInExternalDomainAriaLabel,
+  onClick,
+  style,
+  ...rest
+}: MainLevelProps) => {
+  const menuId = `${id}-menu`;
+  const { autoCollapseOthers, openMainLevels, setMobileMenuOpen, setOpenMainLevels } = useContext(
+    SideNavigationContext,
+  );
+  const activeRef = React.useRef<boolean>(active);
+  const open = openMainLevels.includes(index as number);
+
+  const subLevels = React.Children.map(children, (child) => {
+    if (isValidElement(child) && (child.type as FCWithName).componentName === 'SubLevel') {
+      return cloneElement(child, {
+        mainLevelIndex: index,
+      });
+    }
+    return null;
+  });
+
+  const hasSubLevels = Boolean(subLevels?.length);
+
+  const handleMainLevelClick = (ev: React.MouseEvent<HTMLAnchorElement> | React.MouseEvent<HTMLButtonElement>) => {
+    if (autoCollapseOthers) {
+      // With autoCollapseOther prop only one level can be open at same time
+      setOpenMainLevels([index as number]);
+    } else {
+      // Handle toggle: either remove index if exists or add if it doesn't
+      setOpenMainLevels(xor(openMainLevels, [index]));
+    }
+
+    if (!hasSubLevels) {
+      setMobileMenuOpen(false);
+    }
+
+    if (onClick) {
+      onClick(ev);
+    }
+  };
+
+  React.useEffect(() => {
+    // Ensure that active main level is initially open
+    if (activeRef.current && !open) {
+      if (autoCollapseOthers) {
+        // With autoCollapseOther prop only one level can be open at same time
+        setOpenMainLevels([index as number]);
+      } else {
+        setOpenMainLevels([...openMainLevels, index]);
+      }
+      activeRef.current = false;
+    }
+  }, [activeRef, open, autoCollapseOthers, setOpenMainLevels, openMainLevels, index]);
+
+  return (
+    <li
+      key={id}
+      className={classNames(
+        styles.mainLevel,
+        withDivider && styles.mainLevelWithDivider,
+        active && styles.active,
+        open && styles.open,
+        className,
+      )}
+      style={style}
+    >
+      {hasSubLevels ? (
+        <>
+          <button
+            type="button"
+            aria-current={!!active}
+            aria-label={label}
+            aria-expanded={open}
+            id={id}
+            onClick={handleMainLevelClick}
+          >
+            {icon && <LeftIcon icon={icon} />}
+            <Label label={label} />
+            {hasSubLevels && <RightIcon icon={<IconAngleDown aria-hidden />} className={styles.arrowIcon} />}
+          </button>
+          <ul className={styles.mainLevelListMenu} id={menuId} aria-hidden={!open} aria-labelledby={id}>
+            {subLevels}
+          </ul>
+        </>
+      ) : (
+        <a
+          {...rest}
+          aria-current={active ? 'page' : 'false'}
+          aria-label={external ? `${label} ${openInExternalDomainAriaLabel || 'Siirtyy toiseen sivustoon'}` : label}
```

Comment:
Examples are in English so this additional label probably should be in English too.

Reply:
- [ ] use english label for openInExternalDomainAriaLabel

Reply:
Finnish is probably the most common and default language in Helsinki services though. It might be better to let the default aria-label to be in Finnish language, and change the examples to be in Finnish too. 🤔

Reply:
I fixed it by using English in the story. The default label is still in Finnish.